core/internal/leet/model.go

```diff
@@ -129,43 +282,97 @@ func (m *Model) View() string {
 		return m.renderLoadingScreen()
 	}
 
-	// Show help screen if active
 	if m.help.IsActive() {
-		helpView := m.help.View()
-		statusBar := m.renderStatusBar()
-		// Ensure we use exact height
-		content := lipgloss.JoinVertical(lipgloss.Left, helpView, statusBar)
-		return lipgloss.Place(m.width, m.height, lipgloss.Left, lipgloss.Top, content)
+		return m.renderHelpScreen()
 	}
 
-	var mainView string
+	return m.renderMainView()
+}
+
+// renderHelpScreen renders the help screen.
+func (m *Model) renderHelpScreen() string {
+	helpView := m.help.View()
+	statusBar := m.renderStatusBar()
+
+	content := lipgloss.JoinVertical(lipgloss.Left, helpView, statusBar)
+	return lipgloss.Place(m.width, m.height, lipgloss.Left, lipgloss.Top, content)
+}
+
+// renderMainView renders the main application view.
+func (m *Model) renderMainView() string {
+	layout := m.computeViewports()
+	dims := m.metricsGrid.CalculateChartDimensions(layout.mainContentAreaWidth, layout.height)
+	gridView := m.metricsGrid.View(dims)
+
+	leftWidth := m.leftSidebar.Width()
+	rightWidth := m.rightSidebar.Width()
+
+	// Ensure sidebars don't take up too much space.
+	totalSidebarWidth := leftWidth + rightWidth
+	if totalSidebarWidth >= m.width-10 {
```

Comment:
nit: since this 10 is the same as the one on line 317, a local named `const` would be good